File: M/2027-minimum-moves-to-convert-string/main.go
package main

import "fmt"

func minimumMoves(s string) int {
	idx := 0

	ret := 0
	for idx < len(s) {
		if s[idx] == byte('X') {
			ret++
			idx += 3
		} else {
			idx++
		}
	}
	return ret
}

func main() {
	fmt.Println()
}

File: M/2027-minimum-moves-to-convert-string/main.rs
struct Solution {}

impl Solution {
  pub fn minimum_moves(s: String) -> i32 {
    let mut idx: usize = 0;
    let mut ret: i32 = 0;
    while idx < s.len() {
      if s.as_bytes()[idx] == 'X' as u8 {
        idx += 3;
        ret += 1;
      } else {
        idx += 1;
      }
    }
    ret
  }
}
